### ISSUE TITLE
dataCollectionRuleResourceId in new ALZ policies

### ISSUE THREAD

**Issue by anwather** [Azure/enterprise-azure-policy-as-code#486]
### Discussed in https://github.com/Azure/enterprise-azure-policy-as-code/discussions/485

<div type='discussions-op-text'>

<sup>Originally posted by **arretesting** February 21, 2024</sup>
Recent update (from two days ago) for the ALZ-Root-Default file now requires dataCollectionRuleResourceId. There is a description followed by a link, indicating there must be a DCR in place in Azure Monitor in order to fetch the resource ID from it. Since I had no DCR in place, I created a test DCR in my lab environment, whereas I didn't find the dcrResourceId from anywhere in the Portal. 

Running the following command:

`> Get-AzDataCollectionRule -Name <NameoftheDataCollectionRule> -ResourceGroupName <NameoftheResourceGroup> -SubscriptionId <SubscriptionId> `

led me to believe that the ID shown in the output was the right one. However, while inserting that ID into the parameter and running the Deploy-PolicyPlan script gives me the following error:

```
Assignment error 400 -- {   "error": {     "code": "MissingPolicyParameter",     "message": "The policy assignment 'Deploy-VMSS-Monitoring' is missing the parameter(s) 'dcrResourceId' as
     | defined in the policy definition 'f5bf694c-cca7-4033-b883-3a23327d5485'."   } }
```

What am I missing here? Do I even need to create a DCR or are these created automatically whenever I configure something to monitor?  As VM Insights is to-be retired from Log Analytics and moved to Monitor, is it safe to believe that you will in the following days remove the soon to be deprecated policies from epac? (I believe this has to do with the Deploy diagnostic settings policy).</div>

**Comment by anwather** [comment]
Fixed in v9.1.3

**Comment by anwather** [comment]
Also you will need to create a DCR - the link in the json file links to a bicep file to do this.

**Comment by arretesting** [comment]
Unfortunately, v9.1.3 doesn't fix it for me. Still facing the same error message as described above.

Also, what I did yesterday was to create a DCR from the Portal, just to have one present while running the Deploy-PolicyPlan script. If I look at the json file, it does not link to a bicep file, but to an ARM template, which does exactly that.

For the sake of it, I deleted the DCR I created in the Portal and created a new one through running the following:

`> New-AzResourceGroupDeployment -Name demoDataCollectionRuleDeployment -ResourceGroupName <rgName> -TemplateUri https://raw.githubusercontent.com/Azure/Enterprise-Scale/main/eslzArm/resourceGroupTemplates/dataCollectionRule.json -WorkspaceResourceId "/subscriptions/<subscriptionId>/resourceGroups/<rgName>/providers/Microsoft.OperationalInsights/workspaces/<lawsName>" -WorkspaceLocation "<location>" -userGivenDcrName "<dataCollectionRuleName>"`

Next to that, I removed the old policy and role plans from the output folder just to not have any legacy stuff lying there, I inserted the new resource ID for the DCR (being: /subscriptions/<subscriptionId>/resourceGroups/<rgName>/providers/Microsoft.Insights/dataCollectionRules/<dataCollectionRuleName>) and it still ended up with the Deploy-PolicyPlan script stating that 
**_"The policy assignment 'Deploy-VMSS-Monitoring' is missing the parameter(s) 'dcrResourceId' as defined in the policy definition 'f5bf694c-cca7-4033-b883-3a23327d5485'"._**
